### PR TITLE
Make pallets use construct_runtime

### PR DESCRIPTION
**Pull request by thiolliere**
Related to https://github.com/paritytech/substrate/issues/7949

This PR introduce a few types, and make pallet-assets use construct_runtime.

### CODE REVIEW

File: frame/system/src/lib.rs [hunk 1]
```diff
@@ -1530,3 +1530,15 @@ pub mod pallet_prelude {
 	/// Type alias for the `BlockNumber` associated type of system config.
 	pub type BlockNumberFor<T> = <T as crate::Config>::BlockNumber;
 }
+
+
+/// An unchecked extrinsic type to be used in test.
+pub type MockUncheckedExtrinsic<T, Signature = (), Extra = ()> = generic::UncheckedExtrinsic<
```
Comment (kianenigma) [inline frame/system/src/lib.rs:1536]: better to make these std only?

**Review by kianenigma — APPROVED** [review]

File: frame/support/procedural/src/lib.rs [hunk 2]
```diff
@@ -302,6 +302,11 @@ pub fn decl_storage(input: TokenStream) -> TokenStream {
 /// The population of the genesis storage depends on the order of modules. So, if one of your
 /// modules depends on another module, the module that is depended upon needs to come before
 /// the module depending on it.
+///
+/// # Macro generation
```
Comment (bkchr) [inline frame/support/procedural/src/lib.rs:306]:
```suggestion
/// # Types definitions
```

File: frame/system/src/lib.rs (hunk 1 above)
Comment (bkchr) [reply frame/system/src/lib.rs:1536]: ^^^ + put in some extra file `testing.rs`? Or something like that?

File: frame/support/procedural/src/lib.rs (hunk 2 above)
Comment (dvdplm) [reply frame/support/procedural/src/lib.rs:306]: Should be "Type definitions" I believe.

File: frame/support/procedural/src/lib.rs [hunk 3]
```diff
@@ -302,6 +302,11 @@ pub fn decl_storage(input: TokenStream) -> TokenStream {
...
+///
+/// * The macro generates for each pallet a type alias to their `Module` (or `Pallet`).
```
Comment (dvdplm) [inline frame/support/procedural/src/lib.rs:308]:
```suggestion
/// * The macro generates a type alias for each pallet to their `Module` (or `Pallet`).
```

File: frame/system/src/lib.rs [hunk 4]
```diff
@@ -1530,3 +1530,15 @@ pub mod pallet_prelude {
 	/// Type alias for the `BlockNumber` associated type of system config.
 	pub type BlockNumberFor<T> = <T as crate::Config>::BlockNumber;
 }
+
+
+/// An unchecked extrinsic type to be used in test.
```
Comment (dvdplm) [inline frame/system/src/lib.rs:1535]:
```suggestion
/// An unchecked extrinsic type to be used in tests.
```

File: frame/system/src/lib.rs [hunk 5]
```diff
@@ -1530,3 +1530,15 @@ pub mod pallet_prelude {
...
+pub type MockUncheckedExtrinsic<T, Signature = (), Extra = ()> = generic::UncheckedExtrinsic<
+	<T as Config>::AccountId, <T as Config>::Call, Signature, Extra,
+>;
+
+/// An implementation of `sp_runtime::traits::Block` to be used in test.
```
Comment (dvdplm) [inline frame/system/src/lib.rs:1540]:
```suggestion
/// An implementation of `sp_runtime::traits::Block` to be used in tests.
```

File: frame/system/src/lib.rs (hunk 1 above)
Comment (thiolliere) [reply frame/system/src/lib.rs]: Yes I moved into `mocking.rs` and made it std only

**Review by ascjones — APPROVED** [review]
LGTM

File: frame/assets/Cargo.toml [hunk 6]
```diff
@@ -37,6 +37,7 @@ std = [
 	"codec/std",
 	"sp-std/std",
 	"sp-runtime/std",
+	"pallet-balances/std",
```
Comment (ascjones) [inline frame/assets/Cargo.toml:40]: Actually this should be removed and just remove `default-features = false` from the dev dependency to get `std`

Comment (thiolliere) [reply frame/assets/Cargo.toml]: ah yes, fixed in https://github.com/paritytech/substrate/pull/7950/commits/68307c9ab4477503053a22a26e9015681ba92fad

**Comment by thiolliere** [comment]
bot merge

**Pull request merged**